pipeline/inferrer/aspect_ratio_inferrer/app/encoder.py

```diff
@@ -0,0 +1,26 @@
+from joblib import Parallel, delayed
+
+
+class Encoder:
+    def __init__(self):
+        """
+        Instantiate an encoder which determines the aspect ratio of a given 
+        image, where the aspect ratio (R) = width / height. By definition, R 
+        should always be positive.
+
+        0 < R < 1 : portrait
+        r = 1 : square
+        R > 1 : landscape
+        """
+        self.delayed_process_image = delayed(self.process_image)
+
+    def process_image(self, image):
+        return image.width / image.height
+
+    def __call__(self, images):
+        """
+        process images in parallel
+        """
+        return Parallel(n_jobs=-2)(
```

Comment:
I would hazard a guess that this will be less performant than a single-threaded version because aspect ratio calculation is so trivial and the overhead of thread management will dominate compute time - in the other inferrers we are doing quite complex numerics which benefit from being parallel